usermods/AHT10_v2/usermod_aht10.h

```diff
@@ -0,0 +1,201 @@
+#pragma once
+
+#include "wled.h"
+#include <AHT10.h>
+
+#define AHT10_SUCCESS 1
+
+class UsermodAHT10 : public Usermod {
+  private:
+    static const char _name[];
+
+    unsigned long _lastLoopCheck = 0;
+    bool _initDone = false;
+
+    // Settings. Some of these are stored in a different format than they're user settings - so we don't have to convert at runtime
+    bool _enabled = false;
+    uint8_t _i2cAddress = AHT10_ADDRESS_0X38;
+    ASAIR_I2C_SENSOR _ahtType = AHT10_SENSOR;
+    uint16_t _checkInterval = 60000;    // milliseconds, user settings is in seconds
+    float _decimalFactor = 100;         // a power of 10 factor. 1 would be no change, 10 is one decimal, 100 is two etc. User sees a power of 10 (0, 1, 2, ..)
+
+    uint8_t _lastStatus = 0;
+    float _lastHumidity = 0;
+    float _lastTemperature = 0;
+
+    AHT10* _aht = nullptr;
+
+    float truncateDecimals(float val) {
+      return roundf(val * _decimalFactor) / _decimalFactor;
+    }
+
+    void initializeAht() {
+      if (_aht != nullptr) {
+        delete _aht;
+      }
+
+      _aht = new AHT10(_i2cAddress, _ahtType);
+
+      _lastStatus = 0;
+      _lastHumidity = 0;
+      _lastTemperature = 0;
+    }
+
+    ~UsermodAHT10() {
+      delete _aht;
+      _aht = nullptr;
+    }
+
+  public:
+    void setup() {
+      initializeAht();
+    }
+
+    void loop() {
+      // if usermod is disabled or called during strip updating just exit
+      // NOTE: on very long strips strip.isUpdating() may always return true so update accordingly
+      if (!_enabled || strip.isUpdating())
+        return;
+
+      // do your magic here
+      unsigned long currentTime = millis();
+
+      if (currentTime - _lastLoopCheck < _checkInterval)
+        return;
+      _lastLoopCheck = currentTime;
+
+      _lastStatus = _aht->readRawData();
+      
+      if (_lastStatus == AHT10_ERROR)
+      {
+        // Perform softReset and retry
+        DEBUG_PRINTLN("AHTxx returned error, doing softReset");
```

Comment:
Please enclose strings with `F()` macro. The exception are short strings used multiple times.